src/modinfo.c

```diff
@@ -2645,21 +2675,26 @@ sr_modinfo_module_data_load_srmon(struct sr_mod_info_s *mod_info, struct sr_mod_
 }
 
 /**
- * @brief Check whether the session has pushed any operational data to this module
+ * @brief Check whether the session has pushed any operational data to this module.
+ *        If data exists, load it from cache if available.
  *
  * @param[in] sess Session to check.
  * @param[in] mod_name module name to search.
+ * @param[out] mod_data Cached module data if available. Must be duplicated before use. Can be NULL if flushed or spent already.
  * @return 1 if there is data, else 0
  */
 static int
-sr_modinfo_sess_has_push_oper_data(sr_session_ctx_t *sess, const char *mod_name)
+sr_modinfo_module_data_cache_load(sr_session_ctx_t *sess, const char *mod_name, struct lyd_node **mod_data)
```

Comment:
I know it is a bit annoying but I would still prefer the `out` parameter to be `const` in this case for clarity. Also, seems the functionality is very similar to `sr_module_oper_data_use_cache` but because of this difference (that data needs to be duplicated) I suppose it can remain the way it is. But perhaps just using a flag to decide between using/duplicating the data would be best (and you can put the duplication in the function).